scripts/gulpfile.factory.js

```diff
@@ -3,22 +3,163 @@ module.exports = function factory({
   className,
   precompile = []
 } = {}) {
+  const { task, src, dest, watch, parallel, series } = require("gulp");
+
+  const browser_support = [
+    "last 2 versions",
+    "Firefox > 51",
+    "iOS > 7",
+    "ie 11"
+  ];
+
+  const paths = {
+    source: "./src",
+    compiled: "./",
+    temp: "./tmp"
+  };
+
+  // Tooling
   const fs = require("fs");
   const path = require("path");
+  const replace = require("gulp-replace");
+  const clean = require("gulp-clean");
+
+  // Rollup
+  const shell = require("gulp-shell");
 
-  const gulp = require("gulp");
+  // JavaScript
+  const banner = require("gulp-banner");
   const rename = require("gulp-rename");
-  const replace = require("gulp-replace");
-  const stripCssComments = require("strip-css-comments");
+
+  // Styles
+  const sass = require("gulp-sass");
+  sass.compiler = require("node-sass");
+
+  const postcss = require("gulp-postcss");
+  const sourcemaps = require("gulp-sourcemaps");
+  const autoprefixer = require("autoprefixer");
+  const cleanCSS = require("gulp-clean-css");
+  const postcssCustomProperties = require("postcss-custom-properties");
+
+  // Markup
   const trim = require("trim");
   const decomment = require("decomment");
-  const sass = require("node-sass");
-  const shell = require("gulp-shell");
-  const banner = require("gulp-banner");
 
-  gulp.task("merge", () => {
-    return gulp
-      .src([`./src/**/*.js`])
+  // Compile the sass into css, compress, autoprefix
+  task("compile:sass", () => {
+    return (
+      src("*.{scss,css}", {
+        cwd: paths.source
+      })
+        .pipe(sourcemaps.init())
+        // Compile the Sass into CSS
+        .pipe(
+          sass({
+            outputStyle: "expanded"
+          }).on("error", sass.logError)
+        )
+        // Adds autoprefixing to the compiled sass
+        .pipe(
+          postcss([postcssCustomProperties(), autoprefixer(browser_support)])
+        )
+        // Write the sourcemap
+        .pipe(sourcemaps.write(paths.compiled))
+        // Output the unminified file
+        .pipe(dest(paths.compiled))
+    );
+  });
+
+  task("minify:css", () => {
+    return (
+      src(["*.css"], {
+        cwd: paths.compiled
+      })
+        // Minify the file
+        .pipe(
+          cleanCSS({
+            compatibility: "ie11"
+          })
+        )
+        // Add the .min suffix
+        .pipe(
+          rename({
+            suffix: ".min"
+          })
+        )
+        // Output the minified file
+        .pipe(dest(paths.compiled))
+    );
+  });
+
+  task("fallback:css", () => {
+    const classRegex = new RegExp(`\.${elementName}__(\w+)(.*){`, "gi");
+    return (
+      src(["*.css"], {
```

Comment:
@mwcz This might be too broad since it's causing it to recompile over and over again.  Either that or we have to do a clean before we can run this.